analyzers/utils/processors/unix.go

```diff
@@ -0,0 +1,84 @@
+package processors
+
+import (
+	"bytes"
+	"errors"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/deepsourcelabs/deepsource-go/analyzers/types"
+)
+
+// UnixProcessor is a processor for unix-formatted strings.
+type UnixProcessor struct{}
```

Comment:
How about creating a generic `RegexProcessor` that asks the users for a pattern, and then creates a report based on that? Something like this:

```go
type RegexProcessor struct {
    Pattern string
}

func (rp *RegexProcessor) Process() (Report, error) {
	// logic
}
```

Usage could be like this:

```go
regexProcessor := RegexProcessor{
	Pattern: 	`(?P<filename>\w+):(?P<line>\d+):(?P<column>\d+):(?P<message>.+)\((?P<issue_code>\w+)\)`
}

a := CLIRunner{
	Processor: regexProcessor,
	Command: "true",
}

a.Run()
a.Processor.Process()
....
```